cli/lib/sigstore/cli/id_token.rb

```diff
@@ -85,5 +87,33 @@ def call
         end
       end
     end
+
+    class Buildkite < IdToken
+      def call
+        logger.debug { "looking for OIDC credentials" }
+        unless ENV["BUILDKITE"]
+          logger.debug { "environment doesn't look like Buildkite; giving up" }
+          return
+        end
+
+        raise AmbientCredentialError, "buildkite-agent executable not found" unless buildkite_agent_found?
+
+        request_token
+      end
+
+      private
+
+      def buildkite_agent_found?
+        _, status = Open3.capture2("which buildkite-agent")
+        status.success?
+      end
+
+      def request_token
```

Comment:
we can delete `buildkite_agent_found?` entirely if `request_token` is made to rescue `Errno::ENOENT` and `Errno::EACCES` and then re-raises `AmbientCredentialError`